pkg/bastion/acl.go

```diff
@@ -46,5 +50,51 @@ func checkACLs(user dbmodels.User, host dbmodels.Host) string {
 	}
 	sort.Sort(byWeight(acls))
 
-	return acls[0].Action
+	return checkACLsHook(aclCheckCmd, acls[0].Action, user, host)
+}
+
+// checkACLsHook executes external command to check ACL and passes following parameters:
+// $1 - SSH Portal action (`allow` or `deny`)
+// $2 - User as JSON string
+// $3 - Host as JSON string
+// External program has to return `allow` or `deny` in stdout.
+func checkACLsHook(aclCheckCmd string, action string, user dbmodels.User, host dbmodels.Host) string {
+	if aclCheckCmd != "" {
```

Comment:
you can improve visibility by replacing this `if alcCheckCmd != "" { ..... all the code` by `if alcCheckCmd == "" { return action }`

Reply:
fixed